File: Editorial/137_Single_Number_II/Java/6_Boolean_Algebra_and_Karnaugh_Map.java
class Solution {
    public int singleNumber(int[] nums) {
        // Count (modulo 3) bits
        int msb = 0, lsb = 0;

        // Process Every Num and update count bits
        for (int num : nums) {
            int new_lsb = (~msb & ~lsb & num) | (lsb & ~num);
            int new_msb = (lsb & num) | (msb & ~num);
            lsb = new_lsb;
            msb = new_msb;
        }

        // Return lsb as answer
        return lsb;
    }
}

File: Editorial/137_Single_Number_II/Go/6_Boolean_Algebra_and_Karnaugh_Map.go
func singleNumber(nums []int) int {
    // Count (modulo 3) bits
    msb, lsb := 0, 0
    // Process Every Num and update count bits
    for _, num := range nums {
        new_lsb := (^msb & ^lsb & num) | (lsb & ^num)
        new_msb := (lsb & num) | (msb & ^num)
        lsb = new_lsb
        msb = new_msb
    }
    // Return lsb as the answer
    return lsb
}
